flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/LocalInputChannel.java

```diff
@@ -338,10 +338,13 @@ void releaseAllResources() throws IOException {
 
     @Override
     void announceBufferSize(int newBufferSize) {
-        checkState(!isReleased, "Channel released.");
-
-        ResultSubpartitionView subpartitionView = checkNotNull(this.subpartitionView);
-        subpartitionView.notifyNewBufferSize(newBufferSize);
+        ResultSubpartitionView view = this.subpartitionView;
```

Comment:
I guess you can move that inside of the `if` block.

Reply:
yes, I guess I can but in this case, I will need to double-check subpartitinView for null because if `releaseAllResources` and `announceBufferSize` will be called in parallel it will be a problem. I know that logically, it is impossible right now to execute these methods in parallel but anyway we don't have such explicit restrictions in the code.

Reply:
Can you post the previous code again? I think you did it before anyway. Correct me if I am wrong, but didn't you do?:
```
ResultSubpartitionView view = this.subpartitionView;
if (!isRelease) {
   ResultSubpartitionView subpartitionView = checkNotNull(view);
   ...
}
```